### PR TITLE
bulk get, transaction exec support

### PR DESCRIPTION
**Pull request by mchmarny**
Adds bulk Get API support outlined in issue #46 and the `ExecuteStateTransaction` implementation

### CODE REVIEW

**Comment by yaron2** [comment]
Wow, nicely done! Very timely too.

**Comment by mchmarny** [comment]
> Wow, nicely done! Very timely too.

Nice enough to approve it for merge?

File: client/client.go [hunk 1]
```diff
@@ -48,30 +48,30 @@ type Client interface {
 	// GetSecret retreaves preconfigred secret from specified store using key.
```
Comment (yaron2) [inline client/client.go:48]: nit: change `retreaves` to `retrieves`. There are other occurrences in this PR

File: Readme.md [hunk 2]
```diff
@@ -67,55 +67,84 @@ ctx := context.Background()
...
     },
 }
-err = client.SaveStateItem(ctx, store, data)
+if err := client.SaveStateItems(ctx, store, item1, item2); err != nil {
+    panic(err)
+}
 ```
 
-Similarly, `StateOptions` exist on the `GetDate` and `DeleteState` methods to support multiple item operations at once:
+Similarly, `GetStateItems` method provides a way to retreaded multiple state items in a single operation:
```
Comment (yaron2) [inline Readme.md:120]: `retreaded` to `retrieve`

File: client/client.go [hunk 3]
```diff
@@ -48,30 +48,30 @@ type Client interface {
...
 	// GetState retreaves state from specific store using default consistency option.
-	GetState(ctx context.Context, store, key string) (out []byte, etag string, err error)
+	GetState(ctx context.Context, store, key string) (item *StateItem, err error)
 
 	// GetStateWithConsistency retreaves state from specific store using provided state consistency.
-	GetStateWithConsistency(ctx context.Context, store, key string, sc StateConsistency) (out []byte, etag string, err error)
+	GetStateWithConsistency(ctx context.Context, store, key string, sc StateConsistency) (item *StateItem, err error)
+
+	// GetStateItems retreaves state for multiple keys from specific store.
+	GetStateItems(ctx context.Context, store string, keys []string, parallelism int32) ([]*StateItem, error)
```
Comment (yaron2) [inline client/client.go:64]: Best to change to `GetBulk` or `GetBulkItems`.

File: go.sum [hunk 4]
```diff
@@ -72,6 +76,7 @@ golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
 golang.org/x/text v0.3.3 h1:cokOdA+Jmi5PJGXLlLllQSgYigAEfHXJAERHVMaCc2k=
 golang.org/x/text v0.3.3/go.mod h1:5Zoc/QRtKVWzQhOtBMvqHzDpF6irO9z98xDceosuGiQ=
 golang.org/x/tools v0.0.0-20180917221912-90fa682c2a6e/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
+golang.org/x/tools v0.0.0-20181030221726-6c7e314b6563/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
```
Comment (youngbupark) [inline go.sum:79]: I can see the multiple references to the same pkgs. Can you run `go mod tidy` ?

Comment (mchmarny) [reply go.sum]: done

**Pull request merged**